src/server/models/pushSubscriptionModel.ts

```diff
@@ -10,48 +12,53 @@ interface PushSubscriptionInfo {
 }
 
 class PushSubscriptionModel {
-  private pushSubscriptions: {[id: string]: PushSubscriptionInfo[]};
+  private firestore: Firestore;
 
   constructor() {
-    this.pushSubscriptions = {};
+    this.firestore = new Firestore();
+  }
+
+  _getSubscriptionCollection(login: string) {
+    const collectionPath = `pushsubscriptions/${login}/subscriptions`;
```

Comment:
How about `this.firestore.collection('users').doc(login).collection('subscriptions')`? Seems a bit more clear what's going on, and avoids any issue where `login` might have a `/secrets/` or something in the string.